src/libraries/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -197,7 +197,15 @@ private static Socket Connect(string serverName, int port, TimeSpan timeout)
             Socket[] sockets = new Socket[2];
 
             CancellationTokenSource cts = new CancellationTokenSource();
-            cts.CancelAfter(timeout);
+            if (isInfiniteTimeout)
+            {
+                cts.CancelAfter(-1);
+            }
```

Comment:
This is a nop. Why call CancelAfter at all in this case?

Reply:
Taking an additional look, there are a variety of related things that should be improved here, e.g. if the timeout is infinite, it shouldn't create a CTS at all, and if it isn't, the CTS should be disposed of when done.

Reply:
`CancelAfter(-1)` stands for infinite timeout which can be used here from my first thought. Of course, I can remove it becauseit seems unnecessary at all. Just wonder that, in which case, we can use `CancelAfter(-1)` since it exists? 
And for the CTS dispose, can't we rely on the gabage collect to recycle the resource?

Reply:
> CancelAfter(-1) stands for infinite timeout

Yes, but that timeout is the only thing that would cause cancellation to be requested.  If there's an infinite timeout, there's no need for the CancellationTokenSource at all, nevermind calling CancelAfter(-1).

> can't we rely on the garbage collector to recycle the resource?

Yes, but when you call CancelAfter (with a non-infinite timeout), you're causing the CancellationTokenSource to create a Timer under the covers.  If you don't dispose of that CTS, the Timer won't know it's no longer needed, and will continue taking up resources until it eventually fires.

Reply:
I see. Thank you for the detailed explanation! I will do the change accordingly.